src/main/scala/mesosphere/marathon/core/launchqueue/impl/LaunchQueueActor.scala

```diff
@@ -264,7 +264,7 @@ private[impl] class LaunchQueueActor(
 
       // Reuse resident instances that are stopped.
       val existingReservedStoppedInstances = await(instanceTracker.specInstances(runSpec.id))
-        .filter(i => i.isReserved && i.state.goal == Goal.Stopped) // resident to relaunch
+        .filter(i => i.hasReservation && i.state.goal == Goal.Stopped) // resident to relaunch
```

Comment:
I don't think this is correct. `isReserved` is defined as `isTerminal && hasReservation`. See my comment above.

Reply:
hmm that's a fair point! because now we would reschedule tasks even before they were killed.